src/nlsolve/nlsolve.jl

```diff
@@ -31,28 +39,25 @@ function nlsolve!(nlsolver::AbstractNLSolver, integrator)
         nlsolver.status = Divergence
         break
       end
-
-      # very slow convergence
-      if ndz * θ^(maxiters - iter) > κ * (1 - θ)
-        nlsolver.status = VerySlowConvergence
-        break
-      end
     end
 
     apply_step!(nlsolver, integrator)
 
     # check for convergence
-    if iter == 1 && ndz < 1e-5
-      nlsolver.status = FastConvergence
-      break
-    end
     iter > 1 && (η = θ / (1 - θ))
-    if η * ndz < κ && (iter > 1 || iszero(ndz) || (isnewton(nlsolver) && !iszero(integrator.success_iter)))
-      nlsolver.status = η < fast_convergence_cutoff ? FastConvergence : Convergence
+    if (iter == 1 && ndz < 1e-5) ||
+        iszero(ndz) ||
+        (iter > 1 && (η >= zero(η) && η * ndz < κ))
+      nlsolver.status = Convergence
       break
     end
   end
 
+  if nlsolver.status == Divergence && !isJcurrent(nlsolver, integrator)
+    nlsolver.status = TryAgain
```

Comment:
same dt and J new W?

Reply:
Not necessarily, it is determined by the `do_newJW` function.